ch_tools/common/commands/object_storage.py

```
@@ -154,102 +234,334 @@ def get_object_storage_space_usage(
+        config["remote_blobs_table_prefix"],
+    ]
+
+    database = config["database"]
+
+    for table_prefix in table_prefixes:
+        try:
+            base_name = f"{table_prefix}{disk_conf.name}"
+            table_pattern = f"{base_name}_%"
+
+            # Get tables matching the prefix pattern
+            tables = list_tables(
+                ctx, database_name=database, table_pattern=table_pattern
+            )
+
+            for table_info in tables:
+                table_name = table_info["name"]
+
+                if match := UNIQUE_TABLE_SUFFIX_REGEX.search(table_name):
+                    table_timestamp_str = match.group(1)
+                    try:
+                        table_timestamp = datetime.strptime(
+                            table_timestamp_str, TIMESTAMP_FORMAT
+                        )
+                        table_timestamp = table_timestamp.replace(tzinfo=timezone.utc)
+
+                        # Delete table if it's older than retention period
+                        if table_timestamp < retention_threshold:
+                            full_table_name = f"{database}.{table_name}"
+                            logging.info(
+                                f"Removing old service table: {full_table_name}"
+                            )
+
+                            # Use appropriate drop method based on table type
+                            if table_prefix == config["remote_blobs_table_prefix"]:
+                                drop_table_on_shard(ctx, full_table_name)
+                            else:
+                                drop_table(ctx, full_table_name)
+
+                    except ValueError:
+                        # Skip tables with invalid timestamp format
+                        logging.warning(
+                            f"Skipping table with invalid timestamp format: {table_name}"
+                        )
+                        continue
+
+        except Exception as e:
```

Should we reraise the exception?